engine/src/main/java/io/zeebe/engine/processing/bpmn/behavior/BpmnStateTransitionBehavior.java

```diff
@@ -82,6 +82,17 @@ public BpmnElementContext transitionToActivating(final BpmnElementContext contex
             context.getRecordValue(),
             ProcessInstanceIntent.ELEMENT_ACTIVATING);
       }
+
+      if (context.getElementInstanceKey() == -1) {
+        // processing ACTIVATE_ELEMENT command and no key was generated before
```

Comment:
I think this just describes the code, but not its "why", so we probably don't need it.